### ISSUE TITLE
[bug]: wallet balance below required reserve

### ISSUE THREAD

**Issue by hieblmi** [lightningnetwork/lnd#7591]
Should this scenario be possible?

```
$ lncli walletbalance

{
    "total_balance": "99711",
    "confirmed_balance": "99711",
    "unconfirmed_balance": "0",
    "locked_balance": "0",
    "reserved_balance_anchor_chan": "100000",
    "account_balance": {
        "default": {
            "confirmed_balance": "99711",
            "unconfirmed_balance": "0"
        }
    }
}
```
LND version `v0.16.0-beta`

**Comment by guggero** [comment]
No, it generally shouldn't. But probably not all code paths (such as the sweeper) fully implement the reserve check, so it's apparently still possible. The main goal of the check is to prevent the user from manually removing all funds from the on-chain wallet if there are still channels that might need to be CPFPd.

I guess the main question is: How did you end up in this state? Can you check your on-chain transactions to see what the last action was that made the balance dip below the reserve?

**Comment by hieblmi** [comment]
I forgot to mention that I used `loopd` to loop in some amount which resulted in the wallet balance to fall below the reserve level.

So I guess loop should prevent undershooting the reserve level. Should I create an issue for this?

**Comment by guggero** [comment]
Ah, that is very useful info. I guess we should not really fix this in loop but rather in the RPC calls that loop uses to send on-chain transactions. So I think this issue is sufficient. Looks like `WalletKit.SendOutputs` is the "offending" RPC that needs to have a check added.

**Comment by hieblmi** [comment]
That make a lot of sense to change it on the lnd side. I'll add it to my todo list if no one else is picking this up.

**Comment by antonilol** [comment]
this will always be possible with a different on chain wallet like Electrum with the same seed. Those wallets don't know about this reserve so will let you send out coins even if your balance will go below the reserve.

**Comment by hieblmi** [comment]
Ah ok, but I think it's still good to at least prevent this scenario from occurring in lnd's interfaces where we can?
